File: core/src/main/java/com/ctrip/ferriswheel/core/action/AddText.java
package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.common.action.Action;
import com.ctrip.ferriswheel.common.text.Text;

public final class AddText extends SheetAction implements Action {
    private Text textData;

    public AddText() {
    }

    public AddText(String sheetName, Text textData) {
        super(sheetName);
        this.textData = textData;
    }

    public Text getTextData() {
        return textData;
    }

    public void setTextData(Text textData) {
        this.textData = textData;
    }
}

File: ui/react/src/action/AddText.ts
import Text from '../model/Text';
import SheetAction from './SheetAction';
import Action from './Action';

class AddText extends SheetAction {
    public text: Text;

    public static deserialize(input: any) {
        const sheetName: string = input.sheetName;
        const text = Text.deserialize(input.text);
        return new AddText(sheetName, text);
    }

    constructor(sheetName: string, text: Text) {
        super(sheetName);
        this.text = text;
    }

    public wrapper(): Action {
        const action = new Action();
        action.addText = this;
        return action;
    }

}

export default AddText;
